### ISSUE TITLE
自定义数据集训练完后，推理时显示的全是coco的类别，自定义的类别一个没有。。。

### ISSUE THREAD

**Issue by Flyooofly** [PaddlePaddle/PaddleDetection#2997]
configs/datasets/coco_detection.yml如下
metric: COCO
num_classes: 2

TrainDataset:
  !COCODataSet
    image_dir: train
    anno_path: annotations/trainbelt.json     //自定义数据集的json
    dataset_dir: dataset/coco/
    data_fields: ['image', 'gt_bbox', 'gt_class', 'is_crowd']

EvalDataset:
  !COCODataSet
    image_dir: val
    anno_path:  annotations/valbelt.json
    dataset_dir: dataset/coco/

TestDataset:
  !ImageFolder
    anno_path: annotations/instances_val2017.json

这是模型的配置：
_BASE_: [
  '../datasets/coco_detection.yml',
  '../runtime.yml',
  './_base_/ppyolov2_r50vd_dcn.yml',
  './_base_/optimizer_365e.yml',
  './_base_/ppyolov2_reader.yml',
]

snapshot_epoch: 8
weights: output/ppyolov2_r101vd_dcn_365e_coco/model_final
pretrain_weights: https://paddledet.bj.bcebos.com/models/pretrained/ResNet101_vd_ssld_pretrained.pdparams

ResNet:
  depth: 101
  variant: d
  return_idx: [1, 2, 3]
  dcn_v2_stages: [3]
  freeze_at: -1
  freeze_norm: false
  norm_decay: 0.

我的数据集
![image](https://user-images.githubusercontent.com/31236750/118216507-36174e80-b4a6-11eb-8c4c-c2c198e81aa6.png)

数据集配置应该没问题，因为之前格式不对的时候会提示找不到coco格式的数据，也会显示在正确的路径下没找到，改好以后就开始下载预训练权重开始训练了，难道时因为预训练权重的问题吗

推理出来自定义的类别一个也没有，，全是coco的类，这是为啥呀，，，训练这么久白训练了5555

**Comment by heavengate** [comment]
你好，推理的时候使用的是`TestDataset` 需要修改一下`TestDataset`里的anno配置，看你配置文件里面没修改
```
TestDataset:
!ImageFolder
anno_path: annotations/instances_val2017.json
```

**Comment by Flyooofly** [comment]
> 你好，推理的时候使用的是`TestDataset` 需要修改一下`TestDataset`里的anno配置，看你配置文件里面没修改
> 
> ```
> TestDataset:
> !ImageFolder
> anno_path: annotations/instances_val2017.json
> ```

你好，感谢回答！
然后我把这里改成和val那里的一样了可以吗，但还是不太懂推理的时候为什么还跟这个有关系呢，我给他需要测试的图片的文件路径和训练过的权重不就好了吗，为啥还要这个呀。
另外，我用ppyolov2-101那个训练了200个epoch,为啥效果很差呀。。我其实是要框人。。但框出来一堆别的东西而且准确率都在0.2作用
还有个问题，
![image](https://user-images.githubusercontent.com/31236750/118289342-d5206280-b507-11eb-90fe-035372aab238.png)

就这种训练时候的输出，Epoch[199][0/7]后面的[0/7]是啥意思啊。。是根据batchsize算的吗？总以为接下来会出现Epoch[0][1/7]。。。
麻烦了！

**Comment by heavengate** [comment]
1. 预测的时候也需要读annotation里的类别信息，程序才知道预测的类别是啥再可视化画图，如果没给默认给的是coco数据集的类别信息
2. 可以排查一下`num_classes`配置是否跟你数据集的配置一致，`0/7`是指每个epoch训练7个iter，就是`数据集样本数/卡数/每卡batch_size`，你这个数据集应该比较小，YOLO系列模型收敛比较慢，可以多训练些轮次试试

**Comment by Flyooofly** [comment]
> 1. 预测的时候也需要读annotation里的类别信息，程序才知道预测的类别是啥再可视化画图，如果没给默认给的是coco数据集的类别信息
> 2. 可以排查一下`num_classes`配置是否跟你数据集的配置一致，`0/7`是指每个epoch训练7个iter，就是`数据集样本数/卡数/每卡batch_size`，你这个数据集应该比较小，YOLO系列模型收敛比较慢，可以多训练些轮次试试

好的谢谢回答
